File: examples/baker-example/src/main/java/webshop/simple/MakePaymentInstance.java
package webshop.simple;

import webshop.simple.ingredients.PaymentInformation;
import webshop.simple.ingredients.ReservedItems;
import webshop.simple.ingredients.ShippingAddress;

public class MakePaymentInstance implements MakePayment {
    @Override
    public MakePaymentOutcome apply(String recipeInstanceId,
                                    ReservedItems reservedItems,
                                    PaymentInformation paymentInformation) {
        return new PaymentSuccessful();
    }
}

File: examples/baker-example/src/main/scala/webshop/webservice/MakePaymentInstance.scala
package webshop.webservice

import cats.effect.IO
import webshop.webservice.CheckoutFlowEvents.MakePaymentOutput
import webshop.webservice.CheckoutFlowIngredients.{PaymentInformation, ReservedItems, ShippingAddress, ShippingOrder}
import webshop.webservice.CheckoutFlowInteractions.MakePayment

import scala.concurrent.Future
import scala.concurrent.duration._
import cats.effect.Temporal

class MakePaymentInstance(implicit timer: Temporal[IO]) extends MakePayment {

  override def apply(processId: String, items: ReservedItems, address: ShippingAddress, payment: PaymentInformation): Future[MakePaymentOutput] = {
      IO.sleep(5.second)
        .map(_ => CheckoutFlowEvents.PaymentSuccessful())
        .unsafeToFuture()
  }
}
